protocol-units/bridge/move-modules/sources/atomic_bridge.move

```diff
@@ -0,0 +1,236 @@
+        let bridge_config = BridgeConfig {
+            moveth_minter: signer::address_of(moveth_minter),
+            bridge_module_deployer: signer::address_of(deployer),
+        };
+        move_to(deployer, bridge_transfer_store);
+        move_to(deployer, bridge_config);
+    }
+
+    public fun lock_bridge_transfer_assets(
+        caller: &signer,
+        initiator: vector<u8>, //eth address
+        bridge_transfer_id: vector<u8>,
+        hash_lock: vector<u8>,
+        time_lock: u64,
+        recipient: address,
+        amount: u64
+    ): bool acquires BridgeTransferStore {
+        let bridge_store = borrow_global_mut<BridgeTransferStore>(signer::address_of(caller));
+        let details = BridgeTransferDetails {
+            recipient,
+            initiator,
+            amount,
+            hash_lock,
+            time_lock: block::get_current_block_height() + time_lock
+        };
+
+        smart_table::add(&mut bridge_store.pending_transfers, bridge_transfer_id, details);
+        event::emit(
+            BridgeTransferAssetsLockedEvent {
+                bridge_transfer_id,
+                recipient,
+                amount,
+                hash_lock,
+                time_lock,
+            },
+        );
+
+        true
+    }
+
+    public fun complete_bridge_transfer(
+        caller: &signer,
+        bridge_transfer_id: vector<u8>,
+        pre_image: vector<u8>
+    ) acquires BridgeTransferStore, BridgeConfig {
+        let config_address = borrow_global<BridgeConfig>(@atomic_bridge).bridge_module_deployer;
+        let bridge_store = borrow_global_mut<BridgeTransferStore>(config_address);
+        let details: BridgeTransferDetails = smart_table::remove(&mut bridge_store.pending_transfers, bridge_transfer_id);
+        // Check secret against details.hash_lock
+        let computed_hash = keccak256(pre_image);
+        assert!(computed_hash == details.hash_lock, 2);
+
+        // Mint moveth tokens to the recipient, caller must me a minter of moveth
+        moveth::mint(caller, details.recipient, details.amount);
+
+        smart_table::add(&mut bridge_store.completed_transfers, bridge_transfer_id, details);
+        event::emit(
+            BridgeTransferCompletedEvent {
+                bridge_transfer_id,
+                pre_image,
+            },
+        );
+    }
+
+    public fun abort_bridge_transfer(
+        caller: &signer,
+        bridge_transfer_id: vector<u8>
+    ) acquires BridgeTransferStore, BridgeConfig {
+        // check that the signer is the bridge_module_deployer
+        assert!(signer::address_of(caller) == borrow_global<BridgeConfig>(signer::address_of(caller)).bridge_module_deployer, 1);
+        let bridge_store = borrow_global_mut<BridgeTransferStore>(signer::address_of(caller));
+        let details: BridgeTransferDetails = smart_table::remove(&mut bridge_store.pending_transfers, bridge_transfer_id);
+        assert!(block::get_current_block_height() > details.time_lock, 2);
```

Comment:
This test fails, apparently due to needing to initialize first.

See https://github.com/aptos-labs/aptos-core/issues/5201 for fixes

Reply:
Soz was moving fast, they were passing and I pushed pre-emptively.